src/components/SpruceForm/FieldTemplates.tsx

```diff
@@ -2,26 +2,90 @@ import styled from "@emotion/styled";
 import Button from "@leafygreen-ui/button";
 import {
   ArrayFieldTemplateProps,
+  FieldProps,
   FieldTemplateProps,
   ObjectFieldTemplateProps,
 } from "@rjsf/core";
 import Icon from "components/Icon";
+import { Unpacked } from "types/utils";
 import { SpruceFormContainer } from "./Container";
 import ElementWrapper from "./ElementWrapper";
 
 // Custom field template that does not render fields' titles, as this is handled by LeafyGreen widgets
 export const DefaultFieldTemplate: React.FC<FieldTemplateProps> = ({
   classNames,
   children,
+  description,
+  fields,
   hidden,
-}) => !hidden && <div className={classNames}>{children}</div>;
+  id,
+  label,
+  schema,
+  uiSchema,
+}) => {
+  const isNullType = schema.type === "null";
+  // fields is incorrectly typed in FieldTemplateProps, so cast to avoid type error
+  const { TitleField } = (fields as unknown) as {
+    TitleField: React.StatelessComponent<Partial<FieldProps>>;
```

Comment:
This method has been deprecated. 
![image](https://user-images.githubusercontent.com/4605522/150816375-2d6202a9-2746-4b51-ac48-95e379b741f7.png)

Also im a little confused on why the library is incorrectly typing this value.  The library thinks this should be an array so why are we treating it as an object?
Are we sure this works? Minnas comment below about the field being undefined makes me question if we are accessing the value properly.

Reply:
The library types it as an array but it really is an object! Here's the result of console logging `fields`:
<img width="1194" alt="image" src="https://user-images.githubusercontent.com/9298431/150820930-0864afd1-8dd6-4a17-9c34-db7f2b13c471.png">
 [This](https://github.com/rjsf-team/react-jsonschema-form/blob/master/packages/core/index.d.ts#L26) instance of `fields` is correctly typed but for some reason the `FieldTemplateProps` property isn't. I also opted to use a StatelessComponent because that is how `Field` is typed by the library ([here](https://github.com/rjsf-team/react-jsonschema-form/blob/master/packages/core/index.d.ts#L152)). I could type it as a ClassComponent if you prefer.

The `undefined` Minna referenced above just had to do with the prop not being passed in every usage of the component, it's not related to this issue.

Reply:
The [Docs indicate](https://react-jsonschema-form.readthedocs.io/en/latest/advanced-customization/custom-templates/#fieldtemplate) that fields  is indeed an array in a FieldTemplate 

The linked instance is a [prop on the Form type](https://react-jsonschema-form.readthedocs.io/en/latest/api-reference/form-props/#fields)  which is a different type of attribute. 

Is it possible we are incorrectly using this component and it is actually not a FieldTemplate but instead another type of field?

Reply:
No, it's passed in via the `FieldTemplate` prop [here](https://github.com/evergreen-ci/spruce/blob/main/src/components/SpruceForm/index.tsx#L35) so the usage is as expected for a FieldTemplate. I'm pretty convinced that this is an issue on the library side, but implemented a different approach to avoid it altogether. Let me know what you think!